core/DependencyInjection/ContainerServiceProvider.php

```diff
@@ -30,6 +34,16 @@ public function __invoke(ContainerConfigurator $configurator): void
     {
         $services = $configurator->services();
 
+        /**
+         * @TODO This needs to be override in the scope of middleware task [ADF-731]
+         */
+        $services->set(ServerRequestInterface::class, ServerRequestInterface::class)
```

Comment:
You are incorporating changes I did before and already released

Reply:
Maybe this needs to be rebased? Some changes seem to be already applied by https://github.com/oat-sa/generis/pull/953/files